test/integration/resourcemanager/nodeagentauthorizer/nodeagentauthorizer_suite_test.go

```diff
@@ -0,0 +1,249 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package nodeagentauthorizer_test
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/go-logr/logr"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	controllerconfig "sigs.k8s.io/controller-runtime/pkg/config"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	metricsserver "sigs.k8s.io/controller-runtime/pkg/metrics/server"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/logger"
+	"github.com/gardener/gardener/pkg/resourcemanager/apis/config"
+	resourcemanagerclient "github.com/gardener/gardener/pkg/resourcemanager/client"
+	"github.com/gardener/gardener/pkg/resourcemanager/webhook/nodeagentauthorizer"
+	"github.com/gardener/gardener/pkg/utils"
+	. "github.com/gardener/gardener/pkg/utils/test/matchers"
+	"github.com/gardener/gardener/test/framework"
+)
+
+func TestNodeAgentAuthorizer(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Test Integration ResourceManager NodeAgentAuthorizer Suite")
+}
+
+const testID = "nodeagentauthorizer-webhook-test"
+
+var (
+	ctx = context.Background()
+	log logr.Logger
+
+	sourceRestConfig *rest.Config
+	sourceEnv        *envtest.Environment
+	sourceClient     client.Client
+
+	targetRestConfig          *rest.Config
+	targetRestConfigNodeAgent *rest.Config
+	targetEnv                 *envtest.Environment
+	targetClient              client.Client
+	targetClientNodeAgent     client.Client
+
+	testRunID     string
+	testNamespace *corev1.Namespace
+
+	machineName       string
+	userNameNodeAgent string
+)
+
+var _ = BeforeSuite(func() {
+	logf.SetLogger(logger.MustNewZapLogger(logger.DebugLevel, logger.FormatJSON, zap.WriteTo(GinkgoWriter)))
+	log = logf.Log.WithName(testID)
+	testRunID = utils.ComputeSHA256Hex([]byte(uuid.NewUUID()))[:16]
+
+	// determine a unique namespace name for test environment
+	testNamespaceName := testID + "-" + testRunID[:8]
+
+	By("Start source environment")
+	seedCRDs := filepath.Join("..", "..", "..", "..", "example", "seed-crds")
+	sourceEnv = &envtest.Environment{
+		CRDInstallOptions: envtest.CRDInstallOptions{
+			Paths: []string{
+				filepath.Join(seedCRDs, "10-crd-machine.sapcloud.io_machines.yaml"),
```

Comment:
Do we need a variable for this?
```suggestion
				filepath.Join("..", "..", "..", "..", "example", "seed-crds","10-crd-machine.sapcloud.io_machines.yaml"),
```